### ISSUE TITLE
Create high-level template for ML documentation

### ISSUE THREAD

**Issue by szabosteve** [elastic/stack-docs#470]
We would like to map out a high-level template for ML feature documentation to keep consistency across the various "chapters" in ML. 

Related issue: https://github.com/elastic/stack-docs/issues/459

**Comment by szabosteve** [comment]
# Conceptual overview

[[ml-<feaure>]]
= <Feature_name>

////
The conceptual overview summarizes the main idea and machinery behind the feature. The aim of this high-level summary is to inform the user about the main functions, to familiarize the domain language, basic terms, and to provide some context about the possible fields of usage (short examples, use cases).

The overview section may contain further sub-sections: mind the appropriate heading levels.
////


# How to / Configuration

////
This section shows how to use / configure the feature or contains information related to the feature that couldn’t be the part of the conceptual overview for some reason.
////

# Examples

[[ml-<feaure>-examples]]
== <feaure> examples
++++
<titleabbrev>Examples</titleabbrev>
++++

////
Examples are detailed scenarios of using the feature to solve a specific problem or to cover a use case. Their aim is to show the capabilities of the feature and give the user a sense of its mechanism.
////


# Limitations

[[ml-<feaure>-limitations]]
== <Feature_name> limitations
++++
<titleabbrev>Limitations</titleabbrev>
++++

//// List and description of known issues, limitations of the feature.

The following limitations and known problems apply to the <version number> release of the <name of the feature>:

[float]
[[<feature>-<reference_to_the_limitation>-limitations]]
=== <Limitation_name> limitation

//// Short description of the limitation.


# API quick reference
[[ml-<feature>-apis]]
== API quick reference

All <feature> endpoints have the following base:

[source,js]
----
/_ml/<base_path>
----
// NOTCONSOLE

//// List of links of the API endpoints


# Troubleshooting

[[ml-<feature>-troubleshooting]]
== Troubleshooting <feature>
++++
<titleabbrev>Troubleshooting</titleabbrev>
++++

//// High-level overview of common problems and answers to frequently asked questions.

Use the information in this section to troubleshoot common problems and find
answers for frequently asked questions.

//// List of topics

**Comment by szabosteve** [comment]
@lcawl @sophiec20 

I used our new feature doc checklist (e.g.: https://github.com/elastic/ml-team/issues/175) and the available ML documentation to compile the first version of the documentation template (see above). 

The parts that are commented out are instructions / support text pieces for the writers about the given section.

I found the `How to / Configuration` section the hardest to formalize. Examples for this kind of content are:
* [Evaluating data frame analytics
](https://www.elastic.co/guide/en/elastic-stack-overview/current/ml-dfanalytics-evaluate.html#ml-dfanalytics-evaluate)
* [How data frame transform checkpoints work](https://www.elastic.co/guide/en/elastic-stack-overview/current/ml-transform-checkpoints.html)
* [Configuring machine learning](https://www.elastic.co/guide/en/elastic-stack-overview/current/ml-configuring.html)

I didn't want to further refine this section because it would be harder to fit the content into the structure, but I'm still thinking about this section. I imagine that we'll iterate the template after we have some test drive with it. OFC, we can expand and improve it anytime we see it is necessary.
